### ISSUE TITLE
scalar literals

### ISSUE THREAD

**Issue by Contentmaudlin** [shux-lang/shux#4]
We also want to have exponential scalar literals with the `e` keyword.

`scalar x = 2.54e-10` etc.

Would this require any changes to the scanner?

**Comment by j-hui** [comment]
Yes, gonna add that in after scanner. But we first have to see what the `float_from_string_literal` Function can handle

**Comment by lucas-schuermann** [comment]
Was this ever tested?

**Comment by j-hui** [comment]
no
